lib/origen/sub_blocks.rb

```diff
@@ -404,6 +423,33 @@ def namespace
 
     private
 
+    # @api private
+    # find the block directory path containing the namespace of options[:class_name]
+    def _find_block_dir(options, current_path = nil, remaining_namespace = nil)
+      current_path ||= Pathname.new("#{Origen.root}/app/blocks")
+      remaining_namespace ||= options[:class_name].split('::')[1..-1].map(&:underscore)
+      current_namespace = remaining_namespace.shift
+      if current_namespace
+        if current_path.join(current_namespace).exist?
+          return _find_block_dir(options, current_path.join(current_namespace), remaining_namespace)
+        elsif current_path.join("derivatives/#{current_namespace}").exist?
+          return _find_block_dir(options, current_path.join("derivatives/#{current_namespace}"), remaining_namespace)
+        elsif current_path.join("sub_blocks/#{current_namespace}").exist?
+          return _find_block_dir(options, current_path.join("sub_blocks/#{current_namespace}"), remaining_namespace)
+        else
+          Origen.log.error "Could not find block dir for namespace #{options[:class_name]}!"
+          fail
+        end
+      else
+        if current_path.join('model.rb').exist?
+          return current_path.to_s
+        else
+          Origen.log.error "Could not find block dir for namespace #{options[:class_name]}!"
```

Comment:
should this say could not find model file for namespace?

Reply:
I had this still stay couldn't find block dir, since it tried to search down to the path that matches the block's namespace and even though the actual issue is it didn't find the model.rb, I'm assuming (maybe incorrectly) that this means it wasn't able to find the correct folder containing the class definition files